lib/client.js

```diff
@@ -361,13 +371,33 @@ Client.prototype.stream = function () {
     args.push(function noop() {});
   }
   args = utils.parseCommonArgs.apply(null, args);
+  // NOTE: the nodejs stream maintains yet another internal buffer 
+  // we rely on the default stream implementation to keep memory 
+  // usage reasonable.
   var resultStream = new types.ResultStream({objectMode: 1});
   this.eachRow(args.query, args.params, args.options, function rowCallback(n, row) {
     resultStream.add(row);
-  }, function (err) {
+  }, function (err,result) {
     if (err) {
       resultStream.emit('error', err);
     }
+    if ( result && result.nextPage ) {
+      // allows for throttling as per the
+      // default nodejs stream implementation 
+      resultStream._valve( function( ) {
+          try {
```

Comment:
Could you use 2 spaces for indentation to follow the style of the rest of the project?